MFVNeutralino/test/predict2v_newsimpl.py

```diff
@@ -0,0 +1,175 @@
+        pred_n2v_propagated_stat_err = pred * math.sqrt((eeffn1v/effn1v)**2 + (eeffn1v/effn1v)**2 + (sum2_erest/(sum_rest**2)))
+        epred = pred_n2v_propagated_stat_err
+        print "pred_n2v_propagated_stat_err %8.3f (%8.3f percent)" % (pred_n2v_propagated_stat_err, (100*pred_n2v_propagated_stat_err/pred))
+    print '%8.0f +- %4.0f %8.3f %8.3f %8.3f %9.3f +- %6.3f %7.1f +- %4.1f  PI: [%5.1f, %5.1f] %7.2f +- %.2f PI: [%4.2f, %4.2f]' % (n1v, en1v, f1, cb, cbbar, pred, epred, n2v, en2v, n2v_poisson[0], n2v_poisson[1], rat, erat, eratl, erath)
+print
+print '%16s %16s %8s %8s %8s %8s %19s %15s %35s' % ('n1v0', 'n1v1', 'f1_0', 'f1_1', 'cb', 'cbbar', 'pred n2v', 'n2v', 'ratio')
+
+for ntk in 'Ntk3or4','Ntk3or5', 'Ntk4or5':
+    tracks = [int(i) for i in ntk if i.isdigit()]
+    ntktot = sum(tracks)
+    f1 = []
+    for i, n in enumerate(tracks):
+        if n == 5:
+            tracks[i] = ''
+            f1.append(fracdict[5]['1v'])
+        else:
+            tracks[i] = 'Ntk%s' % n
+            f1.append(fracdict[n]['1v'])
+    n1v0, en1v0 = get_integral(sel_f.Get('%smfvEventHistosOnlyOneVtx/h_npu' % tracks[0]))
+    n1v1, en1v1 = get_integral(sel_f.Get('%smfvEventHistosOnlyOneVtx/h_npu' % tracks[1]))
+    n2v, en2v = get_integral(sel_f.Get('%smfvEventHistosFullSel/h_npu' % ntk))
+    cb = cdict[ntktot]['cb']
+    cbbar = cdict[ntktot]['cbbar']
+    n2v_poisson = poisson_interval(n2v)
+    effn1v0 = n1v0/sum_n1v
+    eeffn1v0 = effn1v0 * math.sqrt( (en1v0/n1v0)**2 + ((sum2_en1v)/(sum_n1v**2)))
+    effn1v1 = n1v1/sum_n1v
+    eeffn1v1 = effn1v1 * math.sqrt( (en1v1/n1v1)**2 + ((sum2_en1v)/(sum_n1v**2))) 
+    #pred = 2 * n1v0*n1v1 / (npresel * (1-f0) * (f1[0] / (1-f1[0]) + 1) * (f1[1] / (1-f1[1]) + 1)) * ((f1[0] / (1 - f1[0])) * (f1[1] / (1 - f1[1])) * ((1 - f0) / f0) * cb + cbbar)
+    pred = (2*(n1v0*n1v1)/(sum_n1v**2)) * (alpha_nm[ntk]) * sum_rest
+    epred = interval_to_vpme(*propagate_ratio(n1v0 * n1v1, npresel, propagate_product(n1v0, n1v1, en1v0, en1v1), enpresel))[1]
+    rat, erat = interval_to_vpme(*propagate_ratio(n2v, pred, en2v, epred))
+    eratl, erath = n2v_poisson / pred
+    #print '%8.0f +- %4.0f %8.0f +- %4.0f %8.3f %8.3f %8.3f %8.3f %9.3f +- %6.3f %7.1f +- %4.1f  PI: [%5.1f, %5.1f] %7.2f +- %4.2f PI: [%4.2f, %4.2f]' % (n1v0, en1v0, n1v1, en1v1, f1[0], f1[1], cb, cbbar, pred, epred, n2v, en2v, n2v_poisson[0], n2v_poisson[1], rat, erat, eratl, erath)
+
+    if print_pred_n2v_propagated_stat_err :
+        #pred_n2v_propagated_stat_err = (n1v0 * n1v1 / npresel) * (f1[0] * f1[1] / f0) * cb * math.sqrt( 1./(n1v0*f1[0]) + 1./(n1v1*f1[1]) ) + (n1v0 * n1v1 / npresel) * ( (1-f1[0]) * (1-f1[1]) / (1-f0) ) * cbbar * math.sqrt( 1./(n1v0 * (1-f1[0])) + 1./(n1v1 * (1-f1[1]))  )
+        pred_n2v_propagated_stat_err = pred * math.sqrt((eeffn1v0/effn1v0)**2 + (eeffn1v1/effn1v1)**2 + (sum2_erest/(sum_rest**2)))
```

Comment:
Same as above: is it `sum2_erest/(sum_rest**2)` or `(sum2_erest/sum_rest)**2`? Is it possible to consolidate some of this to avoid code duplication / potential bugs affecting one vs. the other?

Reply:
The `sum2_erest` is already the quadratic sum of the error due to each term summing up to the `sum_rest`. So, the `sum2_erest/(sum_rest**2)` is by intention.